### ISSUE TITLE
How to pass an account Id in case of multiple accounts?

### ISSUE THREAD

**Issue by sharadhihosamani** [docusign/mobile-android-sdk#18]
Hi ,

I am using JWT grant for login. I'm able to acquire the access token and Login is also successful.

But when trying to pass a valid Template Id (hosted in the docu-sign portal) to TemplateDelegate's useTemplateOnline I am getting below error: 
This account is not authorised to access the requested template. User trying to access template is not authenticated for the template's account.

There are multiple accounts associated with this user. **And the template is present inside an account id which is not set as default.**

iOS SDK take's account id also as an input for authenticating. Hence, they don't face this issue.

Is there a similar way to set/pass the account id in android while initialising / login docu-sign ? or Any other way to overcome this issue?

Thanks!

**Comment by naveentds** [comment]
Hi @sharadhihosamani 
SDK provides the following API to switch the accounts:
```
val authenticationDelegate = DocuSign.getInstance().getAuthenticationDelegate()
authenticationDelegate.switchAccount()
```

Once you switch the account, you should be able to invoke templates using TemplateDelegate's useTemplateOnline API.
Let me know if that works in your use case.

**Comment by sharadhihosamani** [comment]
Hi @naveentds 

The authenticationDelegate.switchAccount() presents user with a screen(PFA) to choose from different accounts available. 
Is there a way to skip this screen and pass the particular account id via code like in iOS ?

<img width="369" alt="Screenshot 2021-06-09 at 11 24 42 AM" src="https://user-images.githubusercontent.com/22168878/121333920-db530480-c936-11eb-98f4-9ca3ab3527b1.png">

**Comment by sharadhihosamani** [comment]
Hi @naveentds any inputs on the above query?  

Also, when i tried authenticationDelegate.switchAccount() to set a different account(which contains the template). The action was successful and in-fact authDelegate.getLoggedInUser() also returns updated account details. 
But templateDelegate.useTemplateOnline() still throws below exception:

com.docusign.androidsdk.exceptions.DSTemplateException: This User lacks sufficient permissions. User trying to access template does not have template edit permission.

Is there anything else that needs to be done/set before accessing online template after switching account?

Thanks!

**Comment by naveentds** [comment]
Hi @sharadhihosamani 
1. Reg. the error ': com.docusign.androidsdk.exceptions.DSTemplateException: This User lacks sufficient permissions. User trying to access template does not have template edit permission', you may want to set permissions. 
Check out the posts:
https://support.docusign.com/s/question/0D51W00007rC5Xt/tranfer-ownership-of-template-doesnt-give-edit-option
https://support.docusign.com/en/guides/ndse-admin-guide-permission-sets
2. Reg. 'a way to skip this screen and pass the particular account id via code like in iOS for switch account', we will definitely add this to our Roadmap in our SDK and this feature will be available in future releases.

Thanks
Naveen

**Comment by sharadhihosamani** [comment]
Thanks for the clarifications @naveentds 

But isn't only Read access enough for pulling a template from portal to create an envelope and launch a signing session? Just trying to understand why we need to have edit permission, as we are not changing/updating the template and the same template and account details work in iOS.

Thanks,
Sharadhi

**Comment by naveentds** [comment]
@sharadhihosamani Currently, SDK supports switching of accounts which has admin access. We will be adding support for switching of non-admin accounts in future SDK releases. 
In the meanwhile, you can try to make the account you are trying to switch has admin access.

**Comment by sharadhihosamani** [comment]
Thanks for the update @naveentds

**Comment by naveentds** [comment]
@sharadhihosamani  We released SDK 1.4.0. Now, you should be able to use the templates in non-admin accounts as well.  Also, we added support for logging in based on the account Id as well.
